### PR TITLE
[CMSP-42] Fixed issue with empty/zero value for posts_per_page

### LINKED ISSUE #566: Divide by zero warning when posting a query
There is an assumption that there is more than one post to calculate posts_per_page in `class-solrpower-wp-query.php`.  This results in a warning.

```txt
PHP Warning:  Division by zero in /code/wp/wp-content/plugins/solr-power/includes/class-solrpower-wp-query.php on line 246
```

### PR DESCRIPTION
**Pull request by pwtyler**
Set posts_per_page to default to 1 if not set.

Replaces #567
Fixes #566.

### CODE REVIEW

File: includes/class-solrpower-wp-query.php [hunk 1]
```diff
@@ -175,22 +175,26 @@ function posts_request( $request, $query ) {
...
 		}
-		$this->qry = $qry;
-		$offset    = $query->get( 'posts_per_page' ) * ( $the_page - 1 );
-		$count     = $query->get( 'posts_per_page' );
-		$fq        = $this->parse_facets( $query );
-		$sortby    = ( isset( $solr_options['s4wp_default_sort'] ) && ! empty( $solr_options['s4wp_default_sort'] ) ) ? $solr_options['s4wp_default_sort'] : 'score';
-		$order     = ( $query->get( 'order', false ) ) ? strtolower( $query->get( 'order' ) ) : 'desc';
-		$extra     = array();
+
+		$posts_per_page = ( ! $query->get( 'posts_per_page' ) ) ? 1 : $query->get( 'posts_per_page' );
```
Comment (jazzsequence) [inline includes/class-solrpower-wp-query.php:189]: I think this should be set to a more intelligent default if it's not set.

```suggestion
		$posts_per_page = get_option( 'posts_per_page' ) ?? 10;
		$posts_per_page = ( ! $query->get( 'posts_per_page' ) ) ? $posts_per_page : $query->get( 'posts_per_page' );
```

**Review by jazzsequence — CHANGES_REQUESTED** [review]
use `posts_per_page` value from the database as a default rather than hard-coding to 1.

File: includes/class-solrpower-wp-query.php (hunk 1 above)
Comment (pwtyler) [reply includes/class-solrpower-wp-query.php:189]: Not sure if two nested defaults is worse than two ternaries but I didn't like ^ lol
```suggestion
		$posts_per_page = $query->get('posts_per_page', get_option('posts_per_page', 10));
```

**Review by jazzsequence — APPROVED** [review]

**Comment by jazzsequence** [comment]
:shipit:

**Pull request merged**